sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala

```diff
@@ -1453,6 +1455,57 @@ class DataFrameAggregateSuite extends QueryTest
     val df = Seq(1).toDF("id").groupBy(Stream($"id" + 1, $"id" + 2): _*).sum("id")
     checkAnswer(df, Row(2, 3, 1))
   }
+
+  test("SPARK-38237: require all cluster keys for child required distribution") {
```

Comment:
This is same unit test added in https://github.com/apache/spark/pull/35552, credited to @HeartSaVioR. I may change the test query to window query instead of aggregate, to make it more convincing.

Reply:
Thanks! Please feel free to adjust the test case or leverage the test case to deduce additional test cases.